src/Expensify.js

```diff
@@ -114,118 +138,77 @@ class Expensify extends PureComponent {
         migrateOnyx()
             .then(() => {
                 // In case of a crash that led to disconnection, we want to remove all the push notifications.
-                if (!this.isAuthenticated()) {
+                if (!isAuthenticated) {
                     PushNotification.clearNotifications();
                 }
 
-                this.setState({isOnyxMigrated: true});
+                setIsOnyxMigrated(true);
             });
 
-        this.appStateChangeListener = AppState.addEventListener('change', this.initializeClient);
+        appStateChangeListener.current = AppState.addEventListener('change', initializeClient);
 
         // Open chat report from a deep link (only mobile native)
         Linking.addEventListener('url', state => Report.openReportFromDeepLink(state.url));
-    }
 
-    componentDidUpdate() {
-        if (!this.state.isNavigationReady || !this.state.isSplashShown) {
+        return () => {
+            if (!appStateChangeListener.current) { return; }
+            appStateChangeListener.current.remove();
+        };
+    }, [isAuthenticated, reportBootSplashStatus]);
+
+    React.useEffect(() => {
+        if (!isNavigationReady || !isSplashShown) {
             return;
         }
 
-        const shouldHideSplash = !this.isAuthenticated() || this.props.isSidebarLoaded;
+        const shouldHideSplash = !isAuthenticated || props.isSidebarLoaded;
 
         if (shouldHideSplash) {
             BootSplash.hide();
 
-            // eslint-disable-next-line react/no-did-update-set-state
-            this.setState({isSplashShown: false});
+            setIsSplashShown(false);
 
             // If the app is opened from a deep link, get the reportID (if exists) from the deep link and navigate to the chat report
             Linking.getInitialURL().then(url => Report.openReportFromDeepLink(url));
         }
-    }
+    }, [props.isSidebarLoaded, isNavigationReady, isSplashShown, isAuthenticated]);
```

Comment:
I'm not sure about adding `isAuthenticated` here, I think that's the reference to the function and will always remain the same?